src/parachain/loans.ts

```diff
@@ -0,0 +1,625 @@
+import { AccountId } from "@polkadot/types/interfaces";
+import { MonetaryAmount } from "@interlay/monetary-js";
+import { BorrowPosition, CurrencyExt, LoanAsset, LendPosition, TickerToData, LendToken, LoanPosition } from "../types";
+import { AssetRegistryAPI } from "./asset-registry";
+import { ApiPromise } from "@polkadot/api";
+import Big from "big.js";
+import {
+    currencyIdToMonetaryCurrency,
+    decodeFixedPointType,
+    MS_PER_YEAR,
+    decodePermill,
+    newCurrencyId,
+    newMonetaryAmount,
+    storageKeyToNthInner,
+} from "../utils";
+import { InterbtcPrimitivesCurrencyId, LoansMarket } from "@polkadot/types/lookup";
+import { StorageKey, Option } from "@polkadot/types";
+import { TransactionAPI } from "./transaction";
+
+/**
+ * @category Lending protocol
+ */
+
+export interface LoansAPI {
+    /**
+     * Get the lend positions for given account.
+     *
+     * @param accountId the account Id for which to get supply positions
+     * @returns Array of lend positions of account.
+     */
+    getLendPositionsOfAccount(accountId: AccountId): Promise<Array<LendPosition>>;
+
+    /**
+     * Get the borrow positions for given account.
+     *
+     * @param accountId the account Id for which to get borrow positions
+     * @returns Array of borrow positions of account.
+     */
+    getBorrowPositionsOfAccount(accountId: AccountId): Promise<Array<BorrowPosition>>;
+
+    /**
+     * Get all loan assets.
+     *
+     * @returns Array of all assets that can be lent and borrowed.
+     * @remarks Method could be refactored to compute APR in lib if we can get underlyingCurrency/rewardCurrency exchange rate,
+     * but is it safe to assume that exchange rate for btc/underlyingCurrency will be
```

Comment:
it is safe to assume that, yes

Reply:
Good to know! This can be refactored later if there is a need - for now I think the current solution works well enough.